### PR TITLE
feat: show next best grid amount

### PR DESCRIPTION
**Pull request by rando128**
## Description

The feature suggests the amount of money you would need to invest in a coin (as a new grid item or a manual trade) to expect closing the grid at break-event (i.e. without losing money) if the coin price eventually reaches your current sell percentage trigger. 

The amount appears only for coins with active grid trades that have one selling grid.

The calculation is based on the following formula: 

**next best grid quantity** = - qi * (1+(p - pi) / (gain * p)), 

where gain = (triggerSellPercentage - 1)
and:

| **Current grid state** | **Suggested best trade**       |
|------------------------|---------------------------|
| Quantity bought (qi)   | Quantity to determine (q) |
| Last price (pi)        | Current price (p)         |
| Spent amount (ai)      | Amount to spend (a)       |

## Motivation and Context

When a coin goes deeper than expected for too long, I often find myself doing a lot of calculations to figure out the best next grid I would need to add to reactivate trading without losing money.

For new comers, it can be difficult to understand how to setup buy-grid-trades and this feature gives some better insight for grid management

## How Has This Been Tested?

It's been running on my server.

## Screenshots (if appropriate):
Screenshot 2022-11-19 at 11 32 59

### CODE REVIEW

File: public/js/CoinWrapperBuySignal.js [hunk 1]
```diff
@@ -301,6 +301,64 @@ class CoinWrapperBuySignal extends React.Component {
...
+      const nextGridQty = -totalBoughtQty * (1 + ((currentPrice - lastBuyPrice) / (gain * currentPrice)));
+
+      const nextGridAmount = nextGridQty * currentPrice;
+
+      return (
+        <React.Fragment
+            key={'coin-wrapper-buy-next-grid-row-' + symbol}>
+          <div className='coin-info-column coin-info-column-price'>
+            <span className='coin-info-label'>
+              &#62; Next grid amount:
```
Comment (chrisleekr) [inline public/js/CoinWrapperBuySignal.js:331]: So this is a suggested grid amount, not the amount the bot will execute it. 
Can we rename the label to `Suggested next grid amount`?

Comment (rando128) [reply public/js/CoinWrapperBuySignal.js:331]: Sure @chrisleekr. I have edited the PR description to clarify this is only a suggestion.

Comment (rando128) [reply public/js/CoinWrapperBuySignal.js]: @chrisleekr, after more thought, I propose to rename 'Suggested break-even amount', which is really about closing the grid without losing money if the market moves up only the sell.triggerPercentage.

**Review by chrisleekr — APPROVED** [review]

**Comment by chrisleekr** [comment]
@rando128 merged!!!

**Comment by rando128** [comment]
Thx @chrisleekr! Hope it will help better adapt our grids to the market changes.

**Comment by peweb** [comment]
@chrisleekr when could you release this? The last one 0.93 is from Oct. 7

**Comment by chrisleekr** [comment]
@peweb Sorry, I will release now.

**Comment by peweb** [comment]
@rando128 this is quite usefull feature but can we make it even better?

It is currently sugegsting the amount to buy at the current price. What if it could offer a purchase amount at a certain percentage difference from the current price?

When I analize the market I set up the next grid to be executed at say -3.5% from the last buy price. I want to profit of say 0.3% so... how to calculate the amount to buy?

**Comment by rando128** [comment]
Good suggestion. I'll look into it.

**Pull request merged**